src/components/cc-product-list/cc-product-list.js

```diff
@@ -0,0 +1,187 @@
+import { css, html, LitElement } from 'lit';
+import { i18n } from '../../lib/i18n.js';
+import { linkStyles } from '../../templates/cc-link/cc-link.js';
+import '../cc-product-card/cc-product-card.js';
+import '../cc-badge/cc-badge.js';
+import '../cc-icon/cc-icon.js';
+import '../cc-input-text/cc-input-text.js';
+import { ProductsController } from './products-controller.js';
+
+/**
+ * @typedef {import('./cc-product-list.types.js').CategoryData} CategoryData
+ * @typedef {import('lit').PropertyValues<CcProductList>} CcProductListPropertyValues
+ */
+
+/**
+ * A component that displays a list of products in sections and allows to search and filter through it.
+ *
+ * @cssdisplay block
+ */
+export class CcProductList extends LitElement {
+
+  static get properties () {
+    return {
+      categoryDataList: { type: Array },
+      filterCategory: { type: String },
+      filterInput: { type: String },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {CategoryData[]} the list of products in their respective categories.  */
+    this.categoryDataList = [];
+
+    /** @type {string} a string to presearch through the list */
+    this.filterInput = '';
+
+    /** @type {'all'|string} a string to prefilter by a given category */
+    this.filterCategory = 'all';
+
+    /** @type {ProductsController} */
+    this._productsCrtl = new ProductsController(this);
+  }
+
+  /**
+   * @param {string} categoryName
+   */
+  _onBadgeClick (categoryName) {
+    this._productsCrtl.filterCategory(categoryName);
+  }
+
+  /**
+   * @param {Object} event
+   * @param {string} event.detail
+  */
+  _onSearchInput ({ detail: value }) {
+    this._productsCrtl.search(value);
```

Comment:
question: Why not just setting the `filterInput` property and the `willUpdate` will take care of applying the filter?